src/main/java/org/axonframework/springboot/nativex/AxonRuntimeHints.java

```diff
@@ -0,0 +1,238 @@
+ * need to have there constructors and methods available for serialisation. It does a few additional things which are
+ * needed to make it work.
+ *
+ * @author Gerard Klijs
+ * @since 4.8.0
+ */
+public class AxonRuntimeHints implements RuntimeHintsRegistrar {
+
+    private static final String PACKAGE_SEPARATOR = ".";
+    private static final String FOLDER_SEPARATOR = "/";
+
+    @Override
+    public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+        ReflectionHints reflectionHints = hints.reflection();
+        registerGrpcHints(reflectionHints);
+        List<Class<?>> axonClasses = getAxonClasses(classLoader);
+        registerReflectionHints(axonClasses, reflectionHints);
+        Set<Class<?>> apiClasses = new HashSet<>(axonSerializableClasses());
+        addClassesUsedInAggregateConstructors(apiClasses, axonClasses);
+        addClassesUsedInHandlers(apiClasses, axonClasses);
+        apiClasses.forEach(c -> registerForSerialisation(reflectionHints, c));
+        hints.resources().registerPattern("axonserver_download.txt");
+        hints.resources().registerPattern("SQLErrorCode.properties");
+        hints.proxies().registerJdkProxy(Connection.class);
+        hints.proxies().registerJdkProxy(TypeReference.of(
+                "org.axonframework.common.jdbc.UnitOfWorkAwareConnectionProviderWrapper$UoWAttachedConnection"));
+    }
+
+    private void registerGrpcHints(ReflectionHints hints) {
+        hints.registerType(EpollChannelOption.class, MemberCategory.DECLARED_FIELDS);
+    }
+
+    private List<Class<?>> getAxonClasses(ClassLoader classLoader) {
+        return getClasses(classLoader, getRootPackage())
+                .stream()
+                .filter(this::isAxonClass)
+                .toList();
+    }
+
+    private void registerReflectionHints(List<Class<?>> axonClasses, ReflectionHints hints) {
+        hints
+                .registerTypes(axonClasses.stream().map(TypeReference::of).toList(),
+                               TypeHint.builtWith(MemberCategory.PUBLIC_CLASSES,
+                                                  MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+                                                  MemberCategory.INVOKE_PUBLIC_METHODS,
+                                                  MemberCategory.INTROSPECT_DECLARED_METHODS,
+                                                  MemberCategory.INTROSPECT_DECLARED_CONSTRUCTORS,
+                                                  MemberCategory.DECLARED_FIELDS));
+    }
+
+    private void registerForSerialisation(ReflectionHints hints, Class<?> c) {
+        hints.registerType(c,
+                           MemberCategory.INVOKE_PUBLIC_CONSTRUCTORS,
+                           MemberCategory.INVOKE_PUBLIC_METHODS);
+    }
+
+    private List<Class<?>> axonSerializableClasses() {
+        return List.of(SagaEntry.class,
```

Comment:
I am mainly doing cross references with the `AxonNativeHints` class...but in there I see we also include the `AbstractSagaEntry.class`.

Reply:
I thought about it, but it's now deprecated/javax which won't work with Spring Boot 3 anyway. So that seemed like a valid reason not to include it?

Reply:
Aaah good one, you're right. If users can only take advantage of this extension with SB3, we do not have to deal with the javax-versions of our entities.